#! /bin/sh /usr/share/dpatch/dpatch-run
## 03_disable_update_system.dpatch by James Healy <jimmy@deefa.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Disable gem update --system functionality, as it can break things. Users should use apt instead

@DPATCH@
diff -urNad '--exclude=CVS' '--exclude=.svn' '--exclude=.git' '--exclude=.arch' '--exclude=.hg' '--exclude=_darcs' '--exclude=.bzr' libgems-ruby~/lib/rubygems/commands/update_command.rb libgems-ruby/lib/rubygems/commands/update_command.rb
--- libgems-ruby~/lib/rubygems/commands/update_command.rb	2010-02-09 14:58:24.000000000 +0900
+++ libgems-ruby/lib/rubygems/commands/update_command.rb	2010-04-07 19:34:33.000000000 +0900
@@ -48,18 +48,7 @@
     hig = {}
 
     if options[:system] then
-      say "Updating RubyGems"
-
-      unless options[:args].empty? then
-        raise "No gem names are allowed with the --system option"
-      end
-
-      rubygems_update = Gem::Specification.new
-      rubygems_update.name = 'rubygems-update'
-      rubygems_update.version = Gem::Version.new Gem::RubyGemsVersion
-      hig['rubygems-update'] = rubygems_update
-
-      options[:user_install] = false
+      fail "gem update --system is disabled on Debian. RubyGems can be updated using the official Debian repositories by aptitude or apt-get."
     else
       say "Updating installed gems"
 
